### DISCUSSION TITLE
What type of component for a rich content tooltip?

### DISCUSSION THREAD

**Discussion by klingat** [radix-ui/primitives#1818] [Help] [answered]
Hello everyone!

If i wanted to create a component for a shared library that acts like a tooltip (appears on hover) but contains rich content (how a popover might contains images, buttons, etc),  what would you use as the base radix component?

My understanding is tooltips are for hover only, should not contain crucial/important information but more just a hint or clarification. For popovers, they should be used for richer content but appear on trigger of a button (click). 

What if I wanted to have a popover that triggers on hover? In other words, a rich tooltip? Should I extend the popover and add the ability to trigger on hover/mouse enter?

Alternately, should I use HoverCard instead? Although my understanding of HoverCard is that it's to be used purely for showing previews of links. 

Thanks for any help or direction steering,

Cheers! 🌸

**Comment by benoitgrelard** [comment, marked as answer]
Hey @klingat,

> My understanding is tooltips are for hover only, should not contain crucial/important information but more just a hint or clarification. For popovers, they should be used for richer content but appear on trigger of a button (click).

Your understanding is correct!

> What if I wanted to have a popover that triggers on hover? In other words, a rich tooltip? Should I extend the popover and add the ability to trigger on hover/mouse enter?

Unfortunately there isn't really a good interaction model which would allow this without hindering accessibility, as if the interaction is on hover, it won't be accessible to keyboard users / screen reader users.

> Alternately, should I use HoverCard instead? Although my understanding of HoverCard is that it's to be used purely for showing previews of links.

Your understanding is correct again. For the same reason above, `HoverCard` is not an accessible pattern, hence why we explicitly say that it should only be used for previewing content that is **already** available behind a link.

So, all in all, I would encourage you to use a `Popover` to make sure all users are included in the experience.

**Reply by klingat** [reply]
@benoitgrelard Thanks for your response and for confirming my thoughts! 😄
